capsules/src/virtual_i2c.rs

```diff
@@ -68,8 +81,41 @@ impl<'a> MuxI2C<'a> {
                     }
                 });
                 node.operation.set(Op::Idle);
-                self.inflight.set(node);
+                self.i2c_inflight.set(node);
             });
+
+            if self.i2c_inflight.is_none() {
+                // No I2C operation in flight, try SMBus next
+                let mnode = self
+                    .smbus_devices
+                    .iter()
+                    .find(|node| node.operation.get() != Op::Idle);
+                mnode.map(|node| {
+                    node.buffer.take().map(|buf| match node.operation.get() {
+                        Op::Write(len) => {
+                            match self.i2c.smbus_write(node.addr, buf, len) {
+                                Ok(_) => {}
+                                Err(e) => self.command_complete(e.1, e.0),
```

Comment:
One issue is if the `smbus_write` call returns an error, we can't unconditionally call a callback here, since `do_next_op()` is called from the functions in the interface. That would cause a callback in the middle of the original call, which is against the execution model. There would have to be a deferred call before calling `do_next_op()`, as in virtual_uart.

Reply:
Fixed. It now uses deferred calls.